saleor/graphql/account/mutations/staff/customer_create.py

```diff
@@ -38,3 +43,39 @@ class Meta:
                 description="Setting a new password for the account is requested.",
             ),
         ]
+
+    @classmethod
+    @traced_atomic_transaction()
+    def save(cls, info: ResolveInfo, instance, cleaned_input):
+        default_shipping_address = cleaned_input.get(SHIPPING_ADDRESS_FIELD)
+        manager = get_plugin_manager_promise(info.context).get()
+
+        if default_shipping_address:
+            default_shipping_address.save()
+            instance.default_shipping_address = default_shipping_address
+
+        default_billing_address = cleaned_input.get(BILLING_ADDRESS_FIELD)
+
+        if default_billing_address:
+            default_billing_address.save()
+            instance.default_billing_address = default_billing_address
+
+        super().save(info, instance, cleaned_input)
+        if default_billing_address:
+            instance.addresses.add(default_billing_address)
+        if default_shipping_address:
+            instance.addresses.add(default_shipping_address)
+
+        instance.search_document = prepare_user_search_document_value(instance)
+        instance.save(update_fields=["search_document", "updated_at"])
+
+        cls.call_event(manager.customer_created, instance)
+        account_events.customer_account_created_event(user=instance)
+
+        if redirect_url := cleaned_input.get("redirect_url"):
+            cls.process_account_confirmation(
```

Comment:
This is called only for `CustomerCreate`. We can fully extract it from the "Base" and place it here.